staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/controller_ref.go

```diff
@@ -31,9 +31,10 @@ func IsControlledBy(obj Object, owner Object) bool {
 
 // GetControllerOf returns a pointer to a copy of the controllerRef if controllee has a controller
 func GetControllerOf(controllee Object) *OwnerReference {
-	for _, ref := range controllee.GetOwnerReferences() {
-		if ref.Controller != nil && *ref.Controller {
-			return &ref
+	refs := controllee.GetOwnerReferences()
+	for i := range  refs {
```

Comment:
If you meant something like this
```
for i := range  controllee.GetOwnerReferences() {
		if controllee.GetOwnerReferences()[i].Controller != nil && *controllee.GetOwnerReferences()[i].Controller {
			return &controllee.GetOwnerReferences()[i]
		}
	}
	return nil
```

then it's about 5 times slower than what I did:
```
pkg: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1
BenchmarkGetControllerOf-12    	20000000	        75.0 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1	1.707s

```

In addition there is no guarantee about performance of the implementation of GetOwnerReferences() method, it may be copying objects so it's better to call it once.

Let me know if you had something else in mind.